vnext/JSI/Shared/ChakraRuntime.cpp

```diff
@@ -17,8 +17,65 @@ using namespace facebook::react;
 
 namespace Microsoft::JSI {
 
-/*static*/ std::once_flag ChakraRuntime::s_runtimeVersionInitFlag;
-/*static*/ uint64_t ChakraRuntime::s_runtimeVersion = 0;
+namespace {
+
+class HostFunctionProxy {
+ public:
+  HostFunctionProxy(
+      facebook::jsi::HostFunctionType hostFunction,
+      ChakraRuntime &runtime)
+      : m_hostFunction(hostFunction), m_runtime(runtime) {}
+
+  inline const facebook::jsi::HostFunctionType &getHostFunction() const {
+    return m_hostFunction;
+  }
+
+  inline ChakraRuntime &getRuntime() const {
+    return m_runtime;
+  }
+
+ private:
+  const facebook::jsi::HostFunctionType m_hostFunction;
+  ChakraRuntime &m_runtime;
+};
+
+class ArgsConverterForCall {
+ public:
+  ArgsConverterForCall(
+      ChakraRuntime &rt,
+      JsValueRef thisObj,
+      const facebook::jsi::Value *args,
+      size_t count) {
+    JsValueRef *destination = inline_;
+    if (count + 1 > maxStackArgs) {
+      outOfLine_ = std::make_unique<JsValueRef[]>(count + 1);
+      destination = outOfLine_.get();
+    }
+
+    if (thisObj == nullptr) {
+      JsValueRef undefinedValue;
+      JsGetUndefinedValue(&undefinedValue);
+      destination[0] = undefinedValue;
+    } else {
+      destination[0] = thisObj;
+    }
+
+    for (size_t i = 0; i < count; ++i) {
+      destination[i + 1] = rt.valueRef(args[i]);
+    }
+  }
+
+  operator JsValueRef *() {
+    return outOfLine_ ? outOfLine_.get() : inline_;
+  }
+
+ private:
+  constexpr static unsigned maxStackArgs = 8;
+  JsValueRef inline_[maxStackArgs];
+  std::unique_ptr<JsValueRef[]> outOfLine_;
```

Comment:
>outOfLine_ [](start = 32, length = 10)

(nit) why some of the private members are "m_*" and others are "*_"
Please be consistent.

Reply:
There will be more Chakra/ChakraCore JSI refactor/stabilization PRs coming, I'll incrementally clean up these inconsistencies.

---
In reply to: [333116326](https://github.com/microsoft/react-native-windows/pull/3366#discussion_r333116326) [](ancestors = 333116326)